src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp

```diff
@@ -383,6 +393,18 @@ class boss_svala : public CreatureScript
                         case EVENT_RITUAL_DISARM:
                             DoCast(me, SPELL_RITUAL_DISARM);
                             break;
+                        case EVENT_FINISH_RITUAL:
+                            events.SetPhase(NORMAL);
+                            me->SetControlled(false, UNIT_STATE_ROOT);
+                            me->SetDisableGravity(false);
+                            me->SetCanFly(false);
+                            summons.DespawnAll();
+                            me->RemoveByteFlag(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_ALWAYS_STAND | UNIT_BYTE1_FLAG_HOVER);
+                            me->GetMotionMaster()->MoveFall();
+                            DoZoneInCombat();
+                            me->SetReactState(REACT_AGGRESSIVE);
+                            AttackStart(me->SelectNearestPlayer(100.0f));
```

Comment:
what happens with cant find any player in 100 yards? will attack null?
will not crash?

Reply:
Would probably crash.  I'll add an if statement around it.

Reply:
Try this:

```cpp
me->SetReactState(REACT_AGGRESSIVE);
DoZoneInCombat();
```
without AttackStart call, it should make she attack

Reply:
Resolved with 2a02067

Reply:
Didn't see your comment @Keader , sorry! Would that be better than what I did in 2a02067?

Reply:
Most AttackStart() implementations do a null check for the passed in parameter and if they don't they should as that's the de-facto standard. Just check UnitAI::AttackStart() .